NotificationApi/NotificationApi.UnitTests/Controller/Notification/GetTemplateByNotificationTypeTests.cs

```diff
@@ -0,0 +1,70 @@
+using Autofac.Extras.Moq;
+using FluentAssertions;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using NotificationApi.Common;
+using NotificationApi.Contract.Responses;
+using NotificationApi.Controllers;
+using NotificationApi.DAL.Queries;
+using NotificationApi.DAL.Queries.Core;
+using NotificationApi.Domain;
+using NotificationApi.Domain.Enums;
+using NUnit.Framework;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace NotificationApi.UnitTests.Controller.Notification
+{
+    public class GetTemplateByNotificationTypeTests
+    {
+        private AutoMock _mocker;
+
+        private NotificationController _sut;
+
+        [SetUp]
+        public void Setup()
+        {
+            _mocker = AutoMock.GetLoose();
+            _sut = _mocker.Create<NotificationController>();
+        }
+
+        [Test]
+        public async Task Should_get_template_by_notification_type_happy()
+        {
+            foreach (var notificationType in Enum.GetValues(typeof(NotificationType)).OfType<NotificationType>())
+            {
+                // Arrange
+                var template = new Template(Guid.NewGuid(), notificationType, MessageType.Email, "parameters");
+                _mocker.Mock<IQueryHandler>().Setup(x => x.Handle<GetTemplateByNotificationTypeQuery, Template>(It.Is<GetTemplateByNotificationTypeQuery>(y => y.NotificationType == notificationType))).ReturnsAsync(template);
+
+                // Act
+                var result = await _sut.GetTemplateByNotificationType((int)notificationType);
```

Comment:
should pass notification type
i thought method was GetTemplateByNotificationTypeAsync?